### PR TITLE
Text layer equation labels

### PR DESCRIPTION
**Pull request by mwestphall**
- Add PyMuPDF dependency for additional text layer processing capabilities
- Replace the "rightmost element" label detection in equation splitter with a text search for parentheses enclosed numbers via PyMuPDF
- Save equation images at a higher resolution with PyMuPDF
- Move all code for switching the coordinate frame of reference between whole-page and equation group into reaggregate_equations.py
- Add utility endpoint for extracting whole text layer of paper with no cosmos annotations, as well as (probable) external urls within the text layer

### CODE REVIEW

File: cosmos/ingestion/ingest/process/aggregation/reaggregate_equations.py [hunk 1]
```diff
@@ -6,21 +6,75 @@
...
+def _overlaps_with_label(bounds: Bounds, label_bounds: List[Bounds]):
+    """ Return whether a Bounds is largely overlapping with a label """
+    area = _area(bounds)
+    return any(_overlap(bounds, lb) > 0.9 * area for lb in label_bounds)
+
+def find_labels_for_equation(pymu_page: fitz.Page, bounds: Bounds):
+    """ Find the text in the text layer matching `(X.Y)` that is in close proximity 
+    to the equation bounding box """
+    rect = _get_scaled_rect(pymu_page, bounds)
+    # guess which "half" of the page the equation is on, then look for a label
```
Comment (iross) [inline cosmos/ingestion/ingest/process/aggregation/reaggregate_equations.py:65]: Hmm. Did we test this on any of the preprints? They're typically single-column, so I think they might cause headaches here. For example:

image

(from 5ebbf1c0998e17af826e5068)

File: cosmos_service/src/db/db.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 Base.metadata.create_all(engine)
 SessionLocal = sessionmaker(bind=engine)
 
-def get_job_details(job_id: str) -> CosmosSessionJob:
+def get_job_details(job_id: str, complete_only=True) -> CosmosSessionJob:
```
Comment (iross) [inline cosmos_service/src/db/db.py:18]: It looks like this is a short-circuit to quickly get the text layer out + available? Can you document it here?

**Review by iross — COMMENTED** [review]
Looks great for the most part! Just a couple things to address before merve.

File: cosmos_service/src/db/db.py (hunk 2 above)
Comment (mwestphall) [reply cosmos_service/src/db/db.py:18]: Yeah, this lets you get the text layer info out before cosmos processing completes. I'll update the documentation.

File: cosmos/ingestion/ingest/process/aggregation/reaggregate_equations.py (hunk 1 above)
Comment (mwestphall) [reply cosmos/ingestion/ingest/process/aggregation/reaggregate_equations.py:65]: Just confirmed that the correct label is found in these cases, however it frequently omits the "Arithmetic mean = " half

**Pull request merged**